### PR TITLE
[MRG] Add `DomainAndLabelStratifiedSubsampleTransformer` + Fix `DomainStratifiedSubsampleTransformer`

### PR DESCRIPTION
**Pull request by YanisLalou**
- Fix `TypeError: DomainStratifiedSubsampleTransformer.transform() got an unexpected keyword argument 'allow_source'`with `DomainStratifiedSubsampleTransformer` 
- Add `DomainAndLabelStratifiedSubsampleTransformer` to have a transformer preserving the percentage of samples
    for each class and for each sample domain (To prevent issues like: `ValueError: This solver needs samples of at least 2 classes in the data, but the data contains only one class: 1`)

### CODE REVIEW

File: skada/transformers.py [hunk 1]
```diff
@@ -129,6 +132,72 @@ def fit_transform(self, X, y=None, *, sample_domain=None, sample_weight=None):
...
+                self._pack_params(
+                    None, sample_domain=sample_domain, sample_weight=sample_weight
+                ),
+            )
+
+        splitter = StratifiedDomainShuffleSplit(
+            n_splits=1, train_size=self.train_size, random_state=self.rng_
+        )
+
+        train_idx, _ = next(splitter.split(X, y, sample_domain))
```
Comment (antoinecollas) [inline skada/transformers.py:191]: `StratifiedDomainShuffleSplit` only statifies w.r.t. the labels but not the domains right? I'm not sure the stratification is done both on domains and labels.

File: skada/tests/test_transformers.py [hunk 2]
```diff
@@ -89,4 +92,50 @@ def test_DomainStratifiedSubsampleTransformer(da_dataset):
...
+    X_subsampled, y_subsampled, params = transformer.fit_transform(
+        X, y, sample_domain=sample_domain, sample_weight=sample_weight
+    )
+
+    assert X_subsampled.shape == (train_size, X.shape[1])
+    assert y_subsampled.shape[0] == train_size
+    assert params["sample_domain"].shape[0] == train_size
+    assert params["sample_weight"].shape[0] == train_size
+    # check stratification
+    assert sum(params["sample_domain"] == 1) == train_size // 2
```
Comment (antoinecollas) [inline skada/tests/test_transformers.py:120]: you should better check the stratification, i.e. compute frequency per (domain, label)

File: skada/transformers.py (hunk 1 above)
Comment (YanisLalou) [reply skada/transformers.py:191]: `StratifiedDomainShuffleSplit` already stratifies w.r.t labels and domains:
```
This cross-validation object returns stratified randomized folds.
The folds are made by preserving the percentage of samples
for each class and for each sample domain.
```

File: skada/tests/test_transformers.py (hunk 2 above)
Comment (YanisLalou) [reply skada/tests/test_transformers.py]: Done

**Comment by antoinecollas** [comment]
Then, we have to be consistent with the splitters. IMO, we have to rename the subsamplers as:
- DomainStratifiedSubsampleTransformer --> DomainSubsampleTransformer
- DomainAndLabelStratifiedSubsampleTransformer --> StratifiedDomainSubsampleTransformer

**Comment by antoinecollas** [comment]
BTW, we can even get rid of 'Transformer':
- DomainStratifiedSubsampleTransformer --> DomainSubsampler
- DomainAndLabelStratifiedSubsampleTransformer --> StratifiedDomainSubsampler

**Comment by YanisLalou** [comment]
I agree

**Review by antoinecollas — APPROVED** [review]

**Pull request merged**